### PR TITLE
feat(lint): impl lint about use first() instead of get(0)

### LINKED ISSUE #8851: New lint: use `slice.first()` instead of `slice.get(0)`
### What it does

Detects `slice.get(0)` and suggests replacement with `slice.first()`.

### Lint Name

slice_get_zero

### Category

style

### Advantage

No magic numbers.

### Drawbacks

`first()` is longer than `get(0)` by 1 character.

### Example

```rust
let v = vec![0, 1, 2];
println!("{}", vec.get(0));
```

Could be written as:

```rust
let v = vec![0, 1, 2];
println!("{}", vec.first());
```

### PR DESCRIPTION
**Pull request by kyoto7250**
close #8851 

This PR adds new lint about considering replacing .get(0) with .first().

Thank you in advance.

changelog: adds new lint [`get_first`] to consider replacing .get(0) with .first()

### CODE REVIEW

File: clippy_lints/src/get_first.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    pub GET_FIRST,
+    style,
+    "Using `x.get(0)` when `x.first()` is simpler"
+}
+declare_lint_pass!(GetFirst => [GET_FIRST]);
+
+impl<'tcx> LateLintPass<'tcx> for GetFirst {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'_>) {
+        if_chain! {
+            if let hir::ExprKind::MethodCall(_, expr_args, _) = &expr.kind;
```
Comment (llogiq) [inline clippy_lints/src/get_first.rs:43]: Using
```suggestion
            if let hir::ExprKind::MethodCall(_, [struct_calling_on, method_arg], _) = &expr.kind;
```
would make the `expr_args.len() == 2` check and the `expr_args.get(_)` checks obsolete.

**Review by llogiq — COMMENTED** [review]
This is already looking quite good. I only have one small suggestion.

File: clippy_lints/src/get_first.rs (hunk 1 above)
Comment (kyoto7250) [reply clippy_lints/src/get_first.rs]: Fixed in [this commit](https://github.com/rust-lang/rust-clippy/pull/8882/commits/d0f93c12a2a6f0d693507ff6325fed62720e616b).

**Comment by kyoto7250** [comment]
@llogiq 

I rebased to the latest master and committed your suggestion.

**Comment by llogiq** [comment]
Thank you!

@bors r+

**Pull request merged**